examples/CustomEntitlementComputationSample/app/src/main/java/com/revenuecat/sample/main/MainScreen.kt

```diff
@@ -0,0 +1,183 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.revenuecat.purchases.Package
+import com.revenuecat.sample.utils.findActivity
+
+@Composable
+fun MainScreen() {
+    val viewModel: MainViewModel = viewModel()
+    val uiState = viewModel.uiState.collectAsState()
+
+    if (uiState.value.shouldStartSwitchingUser) {
+        Dialog(onDismissRequest = { viewModel.resetSwitchUserProcess() }) {
+            SwitchUserDialog(viewModel)
+        }
+    }
+
+    uiState.value.displayErrorMessage?.let { errorMessage ->
+        Toast.makeText(
+            LocalContext.current,
+            errorMessage,
+            Toast.LENGTH_SHORT,
+        ).show()
+        viewModel.resetErrorMessage()
+    }
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = {
+                    Text(
+                        text = "Custom Entitlements Computation App",
+                        letterSpacing = 1.sp,
+                        fontWeight = FontWeight.Black,
+                    )
+                },
+                backgroundColor = Color.Transparent,
+                elevation = 0.dp,
+            )
+        },
+    ) {
+        Box(modifier = Modifier.padding(it)) {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(color = Color.White),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Center,
+            ) {
+                Button(
+                    onClick = { viewModel.initiateSwitchUserProcess() },
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(32.dp),
+                ) {
+                    Text(text = "Switch user")
+                }
+                uiState.value.offerings?.current?.availablePackages?.first()?.let { rcPackage ->
+                    PackageButton(
+                        rcPackage,
+                        "Purchase first offering",
+                    ) { activity, packageToPurchase ->
+                        viewModel.purchasePackage(activity, packageToPurchase)
+                    }
+                }
+                Column(
+                    modifier = Modifier
+                        .weight(1f, true)
+                        .verticalScroll(rememberScrollState())
+                        .padding(32.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.Center,
+                ) {
+                    Box(
+                        contentAlignment = Alignment.TopStart,
+                    ) {
+                        Text(
+                            text = uiState.value.currentCustomerInfo?.rawData?.toString(4)
```

Comment:
Ideally instead of showing raw values here, we'd make it a LazyColumn of customerInfo values over time. 
This part of the app is meant to illustrate how and when customerInfo changes. 
Then when tapping it'd take you to the raw data like we do here

Reply:
it can be a separate PR though

Reply:
oh I see. An old school RecyclerView of updates. Sounds like a necessary improvement. I will start working on that in another branch yes